ostd/src/mm/page_table/node.rs

```diff
@@ -478,20 +479,29 @@ where
 
             // Update the child count.
             if pte.is_none() {
-                // SAFETY: Here we have an exclusive access to the page.
-                unsafe { self.page.meta_mut().nr_children -= 1 };
+                self.meta_mut().nr_children -= 1;
             }
         } else if let Some(e) = pte {
             // SAFETY: This is safe as described in the above branch.
             unsafe { (self.as_ptr() as *mut E).add(idx).write(e) };
-            // SAFETY: Here we have an exclusive access to the page.
-            unsafe { self.page.meta_mut().nr_children += 1 };
+
+            self.meta_mut().nr_children += 1;
         }
     }
 
     fn as_ptr(&self) -> *const E {
         paddr_to_vaddr(self.start_paddr()) as *const E
     }
+
+    fn meta(&self) -> &PageTablePageMetaInner {
+        // SAFETY: Here we have an exclusive access to the page.
```

Comment:
It is not quite appropriate that a reference to the node guard can be regarded as an exclusive access. The code here is sound though.

Reply:
Yes, the guard is exclusive, but the reference is not. The wording here is incorrect.